### PR TITLE
Fixed #32226 -- Fixed JSON format of QuerySet.explain() on PostgreSQL.

### PR DESCRIPTION
**Pull request by whtsky**
Use `json.dumps` to format explain's return when explain format is json.

### CODE REVIEW

**Review by pope1ni — CHANGES_REQUESTED** [review]

**Comment by whtsky** [comment]
Hi, 

I've updated the changes based on suggestions, added myself to the `AUTHORS` and signed the CLA.

File: tests/queries/test_explain.py [hunk 1]
```diff
@@ -30,6 +48,10 @@ def test_basic(self):
                         result = queryset.explain(format=format)
                         self.assertTrue(captured_queries[0]['sql'].startswith(connection.ops.explain_prefix))
                         self.assertIsInstance(result, str)
+                        if format == 'json':
+                            self.assertTrue(is_json_string(result))
+                        elif format == 'xml':
+                            self.assertTrue(is_xml_string(result))
```
[the remark this answers is not in the document]
Comment (tim-mccurrach) [reply tests/queries/test_explain.py]: @pope1ni @whtsky I wonder if something like the following would be more explicit:
```
if format == 'json':
    try:
        json.loads(result)
    except json.JSONDecodeError:
        self.fail("explain_query result is not valid JSON")
```
and likewise for xml.

Perhaps I am being overly pernickety here, but I see a difference between a test raising an exception, and a test failing. The latter being more specific to the thing the test is checking, as well as giving a more helpful error message.

Please feel free to disregard if either of you disagree though.

**Review by tim-mccurrach — APPROVED** [review]
Looks good to me otherwise though :)

I'll move the ticket along

File: tests/queries/test_explain.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (felixxm) [reply tests/queries/test_explain.py]: I agree with Tim, it can be confusing what we are trying to test here without `self.fail()`.

File: tests/queries/test_explain.py [hunk 2]
```diff
@@ -30,6 +32,12 @@ def test_basic(self):
                         result = queryset.explain(format=format)
                         self.assertTrue(captured_queries[0]['sql'].startswith(connection.ops.explain_prefix))
                         self.assertIsInstance(result, str)
+                        if format == 'json':
+                            # Parse the result to check it is valid JSON.
+                            json.loads(result)
+                        elif format == 'xml':
+                            # Parse the result to check it is valid XML.
+                            xml.etree.ElementTree.fromstring(result)
```
Comment (felixxm) [inline tests/queries/test_explain.py:40]: Please move this to a separate commit as an extra test coverage.

Comment (whtsky) [reply tests/queries/test_explain.py:40]: Done

File: tests/queries/test_explain.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (whtsky) [reply tests/queries/test_explain.py]: Done

[the remark this answers is not in the document]
Comment (pope1ni) [reply tests/queries/test_explain.py]: Yup. Happy with this. Was originally mulling over `self.fail(…)`.

**Comment by felixxm** [comment]
@whtsky Thanks :+1: Welcome aboard :boat: 

I pushed small edits.
